internal/myks/globe.go

```diff
@@ -398,3 +398,22 @@ func (g *Globe) getSyncTools() []SyncTool {
 	}
 	return syncTools
 }
+
+func (g *Globe) GetEnvs() map[string]*Environment {
+	return g.environments
+}
+
+func (g *Globe) AddBaseDirToEnvAppMap(envSearchPathToAppMap EnvAppMap) EnvAppMap {
+	envAppMap := EnvAppMap{}
+	for envPath, val := range envSearchPathToAppMap {
+		envAppMap[g.AddBaseDirToEnvPath(envPath)] = val
+	}
+	return envAppMap
+}
+
+func (g *Globe) AddBaseDirToEnvPath(envName string) string {
+	if strings.HasPrefix(envName, g.EnvironmentBaseDir) {
+		return envName
+	}
```

Comment:
This should ensure that we correctly process environment names that start with the same substring as the base dir (e.g. `envsio`, `envstarter`, etc.). 

```suggestion
	if strings.HasPrefix(envName, g.EnvironmentBaseDir+filepath.Separator) {
		return envName
	}
```

(This is not tested)